include/convex_bodies/hpolytope.h

```diff
@@ -638,6 +639,213 @@ class HPolytope {
     }
 
 
+    //------------------------------oracles for exponential sampling---------------//////
+
+    // compute intersection points of a ray starting from r and pointing to v
+    // with polytope discribed by A and b
+    std::pair<NT, int> quadratic_positive_intersect(Point const& r, //current poistion
+                                                    Point const& v, // current velocity
+                                                    VT& Ac, // the product Ac where c is the bias vector of the exponential distribution
+                                                    NT const& T, // the variance of the exponential distribution
+                                                    VT& Ar, // the product Ar 
+                                                    VT& Av, // the product Av
+                                                    int& facet_prev) const //the facet that the trajectory hit in the previous reflection
+    {
+        NT lamda = 0;
+        NT lamda2 =0;
+        NT lamda1 =0;
+        NT alpha;
+        NT min_plus  = std::numeric_limits<NT>::max();
+        VT sum_nom;
+        int m = num_of_hyperplanes();
+        int facet = -1;
+
+        Ar.noalias() = A * r.getCoefficients();
```

Comment:
the two functions `quadratic_positive_intersect` differ only on how `Ar` is computed. Should you combine them?

Reply:
I implemented a new function `get_positive_quadratic_root`. Both oracles are using this function now.